### ISSUE TITLE
"Error reading metadata from cache" and app crash even though space is not full

### ISSUE THREAD

**Issue by rugk** [stevesoltys/backup#531]
I saw https://github.com/seedvault-app/seedvault/issues/213, but somehow I am also having this exception on a device, where the space is not full (anymore).

## STR

Start app.

**What happens:** Crashes.
**What should happen:** No crash.

## Exception

```
FATAL EXCEPTION: main
Process: com.stevesoltys.seedvault, PID: 26499
java.lang.AssertionError: Error reading metadata from cache
at com.stevesoltys.seedvault.metadata.MetadataManager.getMetadata(MetadataManager.kt:48)
at com.stevesoltys.seedvault.metadata.MetadataManager.getBackupToken(MetadataManager.kt:209)
at com.stevesoltys.seedvault.App.migrateTokenFromMetadataToSettingsManager(App.kt:113)
at com.stevesoltys.seedvault.App.access$migrateTokenFromMetadataToSettingsManager(App.kt:44)
at com.stevesoltys.seedvault.App$onCreate$1.invoke(App.kt:80)
at com.stevesoltys.seedvault.App$onCreate$1.invoke(App.kt:79)
at com.stevesoltys.seedvault.AppKt.permitDiskReads(App.kt:142)
at com.stevesoltys.seedvault.App.onCreate(App.kt:79)
at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1270)
at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6785)
at android.app.ActivityThread.-$$Nest$mhandleBindApplication(Unknown Source:0)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2134)
at android.os.Handler.dispatchMessage(Handler.java:106)
at android.os.Looper.loopOnce(Looper.java:201)
at android.os.Looper.loop(Looper.java:288)
at android.app.ActivityThread.main(ActivityThread.java:7898)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
Caused by: java.io.IOException
at com.stevesoltys.seedvault.metadata.MetadataManager.getMetadata(MetadataManager.kt:45)
... 18 more
```

space used: 39GB of 64GB

## system

Device: Pixel 4 (flame)  
OS: Android 13, CalyxOS 4.7.5 
source: F-Droid  
version: v13-3.3
Build-ID: `TQ2A.230305.008.C1.23407050`

**Comment by rugk** [comment]
Note: That exception is old, I just noticed, so maybe the error is something totally different… 😶

**Comment by grote** [comment]
should be fixed by #614
